### ISSUE TITLE
std::get error from getBankConflictInfo

### ISSUE THREAD

**Issue by liqiangxl** [NVIDIA/Fuser#3188]
when run cpp tests with with `NVFUSER_DUMP=bank_conflict`, getBankConflictInfo(fe.kernel()) triggers error `"std::get: wrong index for variant"` when trying to evaluate index with: `expr_eval.evaluate(ti->index()).as<int64_t>();`

Here is a case:
[ RUN      ] NVFuserTest.FusionSmemDynamicPersistentSoftmax2D_CUDA
unknown file: Failure
C++ exception with description "std::get: wrong index for variant" thrown in the test body.

To reproduce: `NVFUSER_TEST_RANDOM_SEED=1729007123 NVFUSER_TEST_ATEN_RANDOM_SEED=0 nvfuser_tests --gtest_filter='NVFuserTest.FusionSmemDynamicPersistentSoftmax2D_CUDA`

**Comment by liqiangxl** [comment]
Hi @zasdfgbnm , is this a known limitation of `getBankConflictInfo`? I know it is currently used in matmul cases, not sure if can be directly used for non-matmul persistent kernels. I noticed this error while working on #3186

**Comment by zasdfgbnm** [comment]
I don't think this is a known limitation. I expect it would just work, otherwise it is a bug.
